### PR TITLE
Enhancement/1003 scheduler registrar

### PR DESCRIPTION
**Pull request by DnlLrssn**
**Checklist**

- [ ] tests added
- [x] commits conform to the [Commit message format](https://github.com/qlik-oss/gopherciser/blob/master/.github/CONTRIBUTING.md#commit)
- [ ] documentation updated

**Squash commit message**

* Add possibility to register a custom scheduler
* Make some scheduler methods public

**Info**

Optional informative text (not to be included in commit message) relevant to reviewers.

### CODE REVIEW

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -41,80 +42,109 @@ type (
...
 		// TimeBuf add wait time in between iterations
 		TimeBuf TimeBuffer `json:"iterationtimebuffer" doc-key:"config.scheduler.iterationtimebuffer"`
 		// InstanceNumber used to ensure different randomizations when running script in multiple different instances
 		InstanceNumber uint64 `json:"instance" doc-key:"config.scheduler.instance"`
 		// ReconnectSettings settings for re-connecting websocket on unexpected disconnect
 		ReconnectSettings session.ReconnectSettings `json:"reconnectsettings" doc-key:"config.scheduler.reconnectsettings"`
 
-		connectionSettings *connection.ConnectionSettings
-		continueOnErrors   bool
+		ConnectionSettings *connection.ConnectionSettings `json:"-"`
```
Comment (SeBBBe) [inline scheduler/scheduler.go:58]: I guess we could add a struct tag for ignoring if this needs to be public.

Comment (DnlLrssn) [reply scheduler/scheduler.go:58]: hmm strange json:"-" already means ignore for json

File: scheduler/scheduler.go [hunk 2]
```diff
@@ -41,80 +42,109 @@ type (
...
+		i++
+		if err := schedEnum.Add(schedulerName, i); err != nil {
+			return nil, err
+		}
 	}
+	return schedEnum, nil
+}
 
-	*value = Type(i)
+// RegisterScheduler register a custom scheduler any existing with same name
```
Comment (atluq) [inline scheduler/scheduler.go:109]: I don't get this sentence

**Review by atluq — APPROVED** [review]

File: scheduler/scheduler.go (hunk 1 above)
Comment (SeBBBe) [reply scheduler/scheduler.go:58]: It doesn't go through json

Comment (SeBBBe) [reply scheduler/scheduler.go:58]: Although I guess maybe we could add something to look for this struct tag and ignore based on that

**Review by SeBBBe — APPROVED** [review]
Working now in QSEST

File: scheduler/scheduler.go [hunk 3]
```diff
@@ -41,80 +42,109 @@ type (
 		PopulateHookData(data map[string]interface{})
 	}
 
+	SchedulerType string
+
 	// Scheduler common core of schedulers
 	Scheduler struct {
 		// SchedType type of scheduler
-		SchedType Type `json:"type" doc-key:"config.scheduler.type"`
+		SchedType SchedulerType `json:"type" doc-key:"config.scheduler.type"`
```
Comment (SeBBBe) [inline scheduler/scheduler.go:50]: Add a DisplayName

File: scheduler/scheduler.go [hunk 4]
```diff
@@ -41,80 +42,109 @@ type (
...
 		// TimeBuf add wait time in between iterations
 		TimeBuf TimeBuffer `json:"iterationtimebuffer" doc-key:"config.scheduler.iterationtimebuffer"`
 		// InstanceNumber used to ensure different randomizations when running script in multiple different instances
 		InstanceNumber uint64 `json:"instance" doc-key:"config.scheduler.instance"`
```
Comment (SeBBBe) [inline scheduler/scheduler.go:54]: This should also have a DisplayName

**Pull request merged**